crates/nu-command/src/conversions/into/filesize.rs

```diff
@@ -148,29 +149,40 @@ fn int_from_string(a_string: &str, span: Span) -> Result<i64, ShellError> {
     let clean_string = no_comma_string.trim();
 
     // Hadle negative file size
-    if let Some(stripped_string) = clean_string.strip_prefix('-') {
-        match stripped_string.parse::<bytesize::ByteSize>() {
+    if let Some(stripped_negative_string) = clean_string.strip_prefix('-') {
+        match stripped_negative_string.parse::<bytesize::ByteSize>() {
             Ok(n) => Ok(-(n.as_u64() as i64)),
-            Err(_) => Err(ShellError::CantConvert {
-                to_type: "int".into(),
-                from_type: "string".into(),
-                span,
-                help: None,
-            }),
+            Err(_) => Err(string_to_into_convert_error(span)),
+        }
+    } else if let Some(stripped_positive_string) = clean_string.strip_prefix('+') {
+        match stripped_positive_string.parse::<bytesize::ByteSize>() {
+            Ok(n)
+                if stripped_positive_string
+                    .chars()
+                    .next()
+                    .map_or(false, |c| c.is_digit(10)) =>
```

Comment:
I think this might be a little simpler.
```suggestion
                if stripped_positive_string.starts_with(|c: char| c.is_ascii_digit()) =>
```

Reply:
Thank you for reviewing! I got it